### PR TITLE
fix: refine impersonlation rfc

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

This PR makes several refinements to the impersonation RFC:

1. Reorganized the structure, add numbers.
2. Refined API endpoint paths and some request/response formats
3. Clarified the `client_id` in the token exchange process
4. Updated the example.

### CODE REVIEW

File: draft/0005-impersonation.md [hunk 1]
```diff
@@ -242,14 +239,16 @@ Content-Type: application/json
 The resource owner API validates the support engineer's request and call Logto's Management API using a machine-to-machine application to issue a subject token for the user.
 
 ```
-POST /api/security/subject-tokens HTTP/1.1
```
Comment (darcyYe) [inline draft/0005-impersonation.md:245]: Seems the API still has the `security/` in the path? Should we remove this part?

File: draft/0005-impersonation.md [hunk 2]
```diff
@@ -242,14 +239,16 @@ Content-Type: application/json
...
-Host: example.logto.app
+POST /api/subject-tokens HTTP/1.1
+Host: example-tenant.logto.app
```
Comment (darcyYe) [inline draft/0005-impersonation.md:243]: Should we use some special mark to indicate that the value of `tenant` could vary? E.g. `example.{tenant}.logto.app`.

File: draft/0005-impersonation.md [hunk 3]
```diff
@@ -269,14 +268,14 @@ The resource owner API returns the subject token to the support engineer.
 
 ### Step 2: Exchange the subject token for an access token
 
-The support engineer exchanges the subject token for an access token that represents the identity of the user:
+The support engineer exchanges the subject token for an access token that represents the identity of the user in the frontend application:
 
 ```
 POST /oidc/token HTTP/1.1
 Host: example.logto.app
```
Comment (darcyYe) [inline draft/0005-impersonation.md:275]: nit: better align the HOST with previous cURL exmaple?

File: draft/0005-impersonation.md [hunk 4]
```diff
@@ -84,39 +85,39 @@ The payload of the access token:
...
 2. `expiresIn`: REQUIRED. The lifetime in seconds of the subject token.
 
-#### Example subject token issuance
+### 4.3 Example subject token issuance
 
 The following example demonstrates a request to issue a subject token and the response. This request is made by a machine-to-machine application that has the necessary permissions to issue subject tokens.
 
 ```
-POST /api/security/subject-tokens HTTP/1.1
+POST /api/subject-tokens HTTP/1.1
```
Comment (darcyYe) [inline draft/0005-impersonation.md:111]: In our API implementation, `security/` is still part of the APIs' path?

File: draft/0005-impersonation.md [hunk 5]
```diff
@@ -6,31 +6,32 @@ Revision: 1
...
-6. `actor_token`: OPTIONAL. A security token that represents the identity of the acting party, it should contain valid `accountId` and have `openid` scope. If present, the access token issued by Logto will contain an additional `act` claim that represents the identity of the acting party.
-7. `actor_token_type`: OPTIONAL. The type of the security token provided in the `actor_token` parameter. The value of this parameter must be `urn:ietf:params:oauth:token-type:access_token`.
+2. `client_id`: REQUIRED. The client identifier of the client application that is making the request, the returned access token will contain this `client_id` claim.
+3. `resource`: OPTIONAL. The resource indicator, the same as other token requests.
+4. `scope`: OPTIONAL. The requested scopes, the same as other token requests.
+5. `subject_token`: REQUIRED. A security token that represents the identity of the party on behalf of whom the request is being made. See the next section for more details.
+6. `subject_token_type`: REQUIRED. The type of the security token provided in the `subject_token` parameter. The value of this parameter must be `urn:ietf:params:oauth:token-type:access_token`.
```
Comment (darcyYe) [inline draft/0005-impersonation.md:30]: nit: should we reorder the parameters and put REQUIRED params before OPTIONAL params?

File: draft/0005-impersonation.md (hunk 1 above)
Comment (wangsijie) [reply draft/0005-impersonation.md:245]: The `security` prefix has been removed in the master.

File: draft/0005-impersonation.md (hunk 2 above)
Comment (wangsijie) [reply draft/0005-impersonation.md:243]: I don't think so, this is just an example, it does not follow the tenant domain format actually.

File: draft/0005-impersonation.md (hunk 5 above)
Comment (wangsijie) [reply draft/0005-impersonation.md:30]: The first 4 params are common in other grants, and then comes the new params.

**Review by gao-sun — APPROVED** [review]

**Review by xiaoyijun — APPROVED** [review]

**Review by darcyYe — APPROVED** [review]

**Pull request merged**